construction skill lets you examine some machines and their circuit boards to learn more about them

Ilysen opened the pull request:
🆑
rscadd: You can now examine many circuit boards to learn some short information about the machine they're used to build. Your Construction skill affects the amount of info you glean, with Trained construction getting the most results.
rscadd: Examining a machine with at least Basic construction skill will tell you about its function.
/🆑

Commit:
Better circuit board examines

Ilysen commented:
also fair warning: my desire to write description blurbs rapidly declined after the two dozenth or so, so ymmv on actual usefulness

SierraKomodo commented on code/game/objects/items/weapons/circuitboards/circuitboard.dm:
```diff
@@ -24,6 +25,30 @@
+			to_chat(user, SPAN_NOTICE("This board can be used for multiple machines. Use a multitool to determine what type of machine that will be created."))
+	if (user.skill_check(SKILL_CONSTRUCTION, SKILL_ADEPT) || isobserver(user))
+		if (req_components.len)
+			to_chat(user, SPAN_NOTICE("It requires the following parts function:"))
```
```suggestion
			to_chat(user, SPAN_NOTICE("It requires the following parts to function:"))
```

SierraKomodo commented on code/game/objects/items/weapons/circuitboards/circuitboard.dm:
```diff
@@ -24,6 +25,30 @@
 		/obj/item/weapon/stock_parts/power/apc/buildable = 1
 	) // unlike the above, this is added to req_components instead of replacing them.
 	var/buildtype_select = FALSE
+	var/machine_name = null // The human-readable name of the machine this board is used to build.
+	var/machine_desc = null // A simple description of what this board is used to build, shown on examine.
```
I feel these vars could be replaced with pulling the name and desc off the `buildtype_select` path instead.

Ilysen replied:
How do ya mean? I believe `buildtype_select` is a bool that determines whether or not the board has multiple variants. Are you referring to `build_path` instead? In that case, would you like, want the vars moved to `/obj/machinery` instead, so that they can be overriden on a per-machine basis?

comma replied:
IMO it would be better to have these descs be on machine, and circtuiboard examine pulling them from there with initial(build_path.name) / initial(build_path.desc)
That way if someone changes machine's functions later on, they can immediately see and update blurb instead of remembering about circuits (which are usually just a copypasted affair)

Ilysen commented:
Commit amended to put `machine_name` and `machine_desc` on `/obj/machine` instead of the circuit boards. Maybe could benefit from a var rename? Unsure - open to input.

Ilysen commented:
Did a big rebase to fix the conflicts caused by the weapons path merge.

SierraKomodo commented on code/game/machinery/Sleeper.dm:
```diff
@@ -11,6 +11,8 @@
 	uncreated_component_parts = null
 	stat_immune = 0
+	machine_name = "sleeper"
+	machine_desc = "Sleepers are high-powered, full-body beds that can synthesize and inject simple chemicals, as well as dialyse substances from a patient's blood and slow down their body functions."
```
```suggestion
	machine_desc = "Sleepers are high-powered, full-body beds that can synthesize and inject simple chemicals, as well as dialyze substances from a patient's blood and slow down their body functions."
```

SierraKomodo commented on code/game/machinery/atmoalter/scrubber.dm:
```diff
@@ -12,6 +12,9 @@
 	power_losses = 150
+	
+	machine_name = "portable scrubber"
+	machine_desc = "Portable scrubbers can be freely moved from place to place in order to draw harmful gases out of the air. Runs on a battery backup; can be connected to atmospherics networks."
```
```suggestion
	machine_desc = "Portable scrubbers can be freely moved from place to place in order to draw harmful gases out of the air. It runs on a battery backup and can be connected to atmospherics networks."
```

SierraKomodo commented on code/game/machinery/vending.dm:
```diff
@@ -16,6 +16,9 @@
 	uncreated_component_parts = null
+	
+	machine_name = "vending machine"
+	machine_desc = "Holds an internal stock of items that can be dispensed on demand or when a charged ID card is swiped, depending on brand."
```
```suggestion
	machine_desc = "Holds an internal stock of items that can be dispensed on-demand or when a charged ID card is swiped, depending on the brand."
```

SierraKomodo commented on code/game/machinery/washing_machine.dm:
```diff
@@ -27,6 +27,9 @@
 	active_power_usage = 150
+	
+	machine_name = "washing machine"
+	machine_desc = "Uses detergent and water to gets your clothes minty fresh. Good for those pesky blood stains! Also decontaminates clothing that has been exposed to toxic elements, as long as detergent is used in the washing process."
```
```suggestion
	machine_desc = "Uses detergent and water to get your clothes minty fresh. Good for those pesky bloodstains! Also decontaminates clothing that has been exposed to toxic elements, as long as detergent is used in the washing process."
```

SierraKomodo commented on code/game/objects/items/weapons/circuitboards/circuitboard.dm:
```diff
@@ -25,6 +26,32 @@
+	if (!user.skill_check(SKILL_CONSTRUCTION, SKILL_BASIC) && !isobserver(user))
+		to_chat(user, "You aren't sure what you can build with this.")
+		return
+	if (build_path)
+		var/obj/machinery/M = build_path
+		var/machine_name = initial(M.machine_name)
+		var/machine_desc = initial(M.machine_desc)
+		if (machine_name && machine_desc)
+			to_chat(user, SPAN_NOTICE("This circuit board is part of \a <b>[machine_name]</b>."))
+			to_chat(user, SPAN_NOTICE(machine_desc))
+			if (buildtype_select)
+				to_chat(user, SPAN_NOTICE("This board can be used for multiple machines. Use a multitool to determine what type of machine that will be created."))
```
This section doesn't appear to actually be functioning in testing - Examining a board with master construction isn't providing the machine name or description.

Ilysen replied:
That's odd. It definitely works on my end. Are you sure you weren't examining something like a power control model or other "generic" circuit board that isn't actually a stock part?

![image](https://user-images.githubusercontent.com/47678781/111688453-928d2300-8801-11eb-941f-058002835acf.png)

SierraKomodo replied:
That was in fact the issue. RIP me.

SierraKomodo commented on code/modules/fabrication/fabricator_food.dm:
```diff
@@ -9,6 +9,27 @@
+
+/obj/machinery/fabricator/replicator/replicator/Initialize()
+	. = ..()
+	if (prob(1))
+		machine_desc = "\
+		<i>\"Fish!\" \
+		<br>\"Today's fish is trout a la creme. Enjoy your meal.\" \
+		<br>\"Fish!\" \
+		<br>\"Today's fish is trout a la creme. Enjoy your meal.\" \
+		<br>\"Fish!\" \
+		<br>\"Today's fish is trout a la creme. Enjoy your meal.\" \
+		<br>\"Fish!\" \
+		<br>\"Today's fish is trout a la creme. Enjoy your meal.\" \
+		<br>\"Fish!\" \
+		<br>\"Today's fish is trout a la creme. Enjoy your meal.\" \
+		<br>\"Fish!\" \
+		<br>\"Today's fish is trout a la creme. Enjoy your meal.\" \
+		<br>\"I will!\" \
+		</i>"
```
While cute, I don't think we need something that floods the chat log like this.

Ilysen replied:
Think it'd be fine if it was something like three lines instead?

SierraKomodo commented on code/modules/mining/drilling/drill.dm:
```diff
@@ -255,6 +257,9 @@
 	interact_offline = 1
+	
+	machine_name = "mining drill brace"
+	machine_desc = "A mobile support strut that provides structure for the head of a mining drill when anchored. Placed on either side of the drill head."
```
```suggestion
	machine_desc = "A mobile support strut that provides support for the head of a mining drill when anchored. Placed on either side of the drill head."
```

SierraKomodo reviewed (changes requested):
May also be worth including the `machine_desc` in a machine's normal examine text as I'm noticing a pattern of these including information that isn't otherwise displayed when looking at the machine itself.

SierraKomodo reviewed (changes requested):
Also, merge conflicts

Ilysen commented:
Addressed the review stuff. I kept the easter egg, but reduced it to two lines; if you want me to remove it entirely, I'll be whiny but I'll still do it.

I also made examining machines display their `machine_desc` on recommendation. I'm worried it might cause some chat spam, but hopefully it'll be fine.

SierraKomodo reviewed (changes requested):
Merge conflicts

Ilysen commented:
Just to make sure this didn't get overlooked, it seems like the conflicts tag didn't get automatically removed.
